storm-server/src/main/java/org/apache/storm/scheduler/Cluster.java

```diff
@@ -664,6 +722,9 @@ private double calculateSharedOffHeapMemory(String nodeId, SchedulerAssignmentIm
 
     private double calculateSharedOffHeapMemory(
         String nodeId, SchedulerAssignmentImpl assignment, ExecutorDetails extra) {
+        if (assignment == null) {
```

Comment:
When is the assignment null???  Why would we try to calculate the shared off heap memory for no assignment???